pkg/storageops/aws/aws.go

```diff
@@ -529,40 +536,46 @@ func (s *ec2Ops) Detach(volumeID string) error {
 func (s *ec2Ops) Snapshot(
 	volumeID string,
 	readonly bool,
-) (*ec2.Snapshot, error) {
+) (interface{}, error) {
 	request := &ec2.CreateSnapshotInput{
 		VolumeId: &volumeID,
 	}
 	return s.ec2.CreateSnapshot(request)
 }
 
-func (s *ec2Ops) DevicePath(vol *ec2.Volume) (string, error) {
+func (s *ec2Ops) DevicePath(v interface{}) (string, error) {
```

Comment:
same here. May be we could just change the API to DevicePath(volumeId string). So that we don't need to pass in templates or the caller doesn't need to create a ec2Volume object. What do you think?

Reply:
Changed API in incremental